EIPS/eip-1238.md

```diff
@@ -0,0 +1,213 @@
+
+## Motivation
+The idea is to have tokens that once assigned cannot be transferred (like reputation) and that can be used by websites, or contracts to make users perform some actions. For example, if a user accumulates paper submissions at conferences, then they can use their paper badges to request grants. It's important that they can never share these badges.
+
+This is the equivalent of a variety of other use cases
+
+- Accumulating badges: degrees from academic institutions, paper publications for conferences, badges that allow for access in a building or in a special chat.
+- Experience points: points accumulated in a game, points accumulated by being honest in a decentralized system for some time or a DAO.
+- Statements: more broadly, any statement that is issued or signed by a contract, a dao, a single user that requires to be on-chain.
+- Subscription: badges can represent the validity of a paid subscription.
+
+Additionally, this standard could be used as a primitive for Soulbound tokens, whereby the recovery of a Soul would be done by obtaining signatures from co-owners of tokens with the same id.
+
+Multiple proposals for non-transferable / Soulbound tokens have been put forth, however this one presents a unique combination of features which includes having multiple token types in one contract (similar to [EIP-1155](./eip-1155.md)), recipient's consent and, optionally, token staking and expiry.
+
+## Specification
+Every contract compliant with this EIP must implement the following interface:
+
+```solidity
+interface IERC1238 is IERC165 {
+    /**
+     * @dev Emitted when `amount` tokens of token type `id` are minted to `to` by `minter`.
+     */
+    event MintSingle(address indexed minter, address indexed to, uint256 indexed id, uint256 amount);
+
+    /**
+     * @dev Equivalent to multiple {MintSingle} events, where `minter` and `to` is the same for all token types.
+     */
+    event MintBatch(address indexed minter, address indexed to, uint256[] ids, uint256[] amounts);
+
+    /**
+     * @dev Emitted when `amount` tokens of token type `id` owned by `owner` are burned by `burner`.
+     */
+    event BurnSingle(address indexed burner, address indexed owner, uint256 indexed id, uint256 amount);
+
+    /**
+     * @dev Equivalent to multiple {BurnSingle} events, where `owner` and `burner` is the same for all token types.
+     */
+    event BurnBatch(address indexed burner, address indexed owner, uint256[] ids, uint256[] amounts);
+
+    /**
+     * @dev Returns the amount of tokens of token type `id` owned by `account`.
+     *
+     * Requirements:
+     *
+     * - `account` cannot be the zero address.
```

Comment:
Why can't the zero address hold tokens? Is that a hard requirement for all implementations? Again, probably worth putting in the Rationale.

Reply:
Added more in the Rationale section.
I'm thinking about the consequences if a balance check is used for gating access and there's a bug and the zero address is passed instead of the user's address...